ethcore/src/executive.rs

```diff
@@ -113,12 +113,20 @@ impl<'a> Executive<'a> {
 	}
 
 	/// Creates `Externalities` from `Executive`.
-	pub fn as_externalities<'_>(&'_ mut self, origin_info: OriginInfo, substate: &'_ mut Substate, output: OutputPolicy<'_, '_>) -> Externalities {
-		Externalities::new(self.state, self.info, self.engine, self.depth, origin_info, substate, output)
+	pub fn as_externalities<'_, T>(&'_ mut self, origin_info: OriginInfo, substate: &'_ mut Substate, output: OutputPolicy<'_, '_>, tracer: &'_ mut T) -> Externalities<'_, T> where T: Tracer {
+		Externalities::new(self.state, self.info, self.engine, self.depth, origin_info, substate, output, tracer)
 	}
 
 	/// This funtion should be used to execute transaction.
 	pub fn transact(&'a mut self, t: &SignedTransaction, options: TransactOptions) -> Result<Executed, Error> {
+		let check = options.check_nonce;
+		match options.tracing {
+			true => self.transact_with_tracer(t, check, ExecutiveTracer::default()),
```

Comment:
> just pull out the final param into the conditional

I wanted, but I can't.

```rust
let t = if options.tracing {
  ExecutiveTracer::default()
else {
  NoopTracer
}
```

This is not valid expression, because of the type difference. And if I Box the tracer, the type information is lost.